Team 5 Homework 1 Submission

jazwu opened the pull request.

qianxichen233 committed:
Setup circleCI and clang-tidy

kyotov commented:
this looks good!
couple of comments --
- it is a small change, why have 4 commits? maybe squash them into one to make reviewing easier.
- include a link to the circleci run that shows the tests running
- did you try to run the rest of the tests in ninja (there are some perf tests for example)...

qianxichen233 commented:
Here is the link for our successful circleci build with default ninja unit test running: https://app.circleci.com/pipelines/circleci/89RxQ8gXukBs6pXwF2XEFo/LBJg2NqgUMKxynB5gJyqDg/10/workflows/6d9b62c4-3a0b-4dac-bc70-a799bca5e251

For the third comment, I didn't notice there is perftest before since it is not mentioned in ninja's README. I just tried to run these perftest and most of them are able to run, but there is one perftest that requires some kind of command line argument (depfile_parser_perftest) and I am not very sure what to pass to it since there doesn't seem to be any open documentation for it, nor there is any existing scripts that run the perftest automatically. Besides, I also saw some fuzzing test related stuff under misc folder, and again there doesn't seem to be any instruction on how to run it. I guess I'll try to figure it out later

kyotov commented:
cool -- seeing what binaries it builds can help figure out what tests are there.
thanks for the link on CI -- it is interesting that it took 15 min to build -- sounds long?
let's see what others got.

qianxichen233 commented:
yeah, the 15 mins are mostly from clang-tidy checks. If I disable clang-tidy, it can build much more faster. Like I am not very what clang-tidy checks we want to enable for shadowdash

kyotov commented:
ah, that makes sense.
clangtidy can be slow indeed.
we should probably split it in a separate pipeline eventually.

kyotov commented:
one thing you can do is break a test on purpose and see how that shows up

kyotov commented:
seems like more integration is needed on the tests side?
![image](https://github.com/user-attachments/assets/8fc90422-fc3f-4363-9a12-9932641de489)

qianxichen233 commented:
> seems like more integration is needed on the tests side? ![image](https://private-user-images.githubusercontent.com/517176/374769612-8fc90422-fc3f-4363-9a12-9932641de489.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg0Mzg4MzQsIm5iZiI6MTcyODQzODUzNCwicGF0aCI6Ii81MTcxNzYvMzc0NzY5NjEyLThmYzkwNDIyLWZjM2YtNDM2My05YTEyLTk5MzI2NDFkZTQ4OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAwOVQwMTQ4NTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04ZTBiNTBjOGM2NTQ5NjJmZDFjNzc1MThmZjE4ZGVhMTcxNjY2ZDY5NWYwYTc1NzNiYzU0NTE2NTNhMGMzNTg5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.8xKWbROojSZVu1BqR87Yqc9nNeqrmzgXzkpsVz6hnxA)

I am not very sure what this is. I don't have any test_sample.py in the repo, looks like this is something CircleCI automatically created in some place

kyotov commented:
there are standard ways that CI parses test resutls.
you can ask your testing framework to produce the right file and then
circleci can consume it.
i can talk a bit about it during class tomorrow.

On Tue, Oct 8, 2024 at 10:01 PM Qianxi Chen ***@***.***>
wrote:

> seems like more integration is needed on the tests side? [image: image]
> <https://private-user-images.githubusercontent.com/517176/374769612-8fc90422-fc3f-4363-9a12-9932641de489.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg0Mzg4MzQsIm5iZiI6MTcyODQzODUzNCwicGF0aCI6Ii81MTcxNzYvMzc0NzY5NjEyLThmYzkwNDIyLWZjM2YtNDM2My05YTEyLTk5MzI2NDFkZTQ4OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAwOVQwMTQ4NTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04ZTBiNTBjOGM2NTQ5NjJmZDFjNzc1MThmZjE4ZGVhMTcxNjY2ZDY5NWYwYTc1NzNiYzU0NTE2NTNhMGMzNTg5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.8xKWbROojSZVu1BqR87Yqc9nNeqrmzgXzkpsVz6hnxA>
>
> I am not very sure what this is. I don't have any test_sample.py in the
> repo, looks like this is something CircleCI automatically created in some
> place
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/nyuoss/ninja-shadowdash-nyu/pull/2#issuecomment-2401122891>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAD6IOGNBJJA3MIMQAEOTF3Z2SE6TAVCNFSM6AAAAABPEPRD26VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDIMBRGEZDEOBZGE>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

qianxichen233 commented:
I figured out what the depfile_parser_perftest is doing (the perftest that expects some command line argument). It is a bench test for parsing dependency files (.d file). I didn't find any existing example .d file in ninja, so do we want to create our own test dependency file, or we just ignore this perftest?

kyotov commented on .circleci/config.yml:
```diff
@@ -0,0 +1,55 @@
+executors:
+  docker-executor:
+    docker:
+      - image: cimg/base:stable
```
you can build your own image for extra credit, but this is ok for this hw

kyotov commented on .circleci/config.yml:
```diff
@@ -0,0 +1,55 @@
+      - run:
+          name: Run Tests
+          command: |
+            ./build-cmake/ninja_test
+            ./build-cmake/build_log_perftest
+            ./build-cmake/canon_perftest
+            ./build-cmake/clparser_perftest
+            ./build-cmake/depfile_parser_perftest
+            ./build-cmake/elide_middle_perftest
+            ./build-cmake/hash_collision_bench
+            ./build-cmake/manifest_parser_perftest
```
maybe these should be separate tests running in parallel?

kyotov replied:
try using ctest

kyotov reviewed (changes requested):
review and address all comments and let me know when it is ready for another look

kyotov commented:
lets also add clangformat

Shihuihuang1103 committed:
create docker image & test parallelism

Use ctest and clang-format

Remove add_executable and add_test

Add other tests to ctest

Fix failed tests

Fix test dir

Adjust tests: depfile_parser, manifest_parser

Set up docker image, use ctest, clang-tidy and parallel testing

qianxichen233 committed:
Merge pull request #1 from qianxichen233/circleci-setup-update

Create docker image & test parallelism

Shihuihuang1103 commented:
Hi Professor, all comments are addressed and this should be ready for a second review. Thank you!
Here is the link to the CircleCI run: [link](https://app.circleci.com/pipelines/circleci/89RxQ8gXukBs6pXwF2XEFo/LBJg2NqgUMKxynB5gJyqDg/37/workflows/0c0145e4-9a61-41e0-8efd-e8a9c87a113f)

Shihuihuang1103 committed:
Add store_test_results to integrate with circleci

Fix no test result found

fix test results dir

Try output test results to circleci

test write to test-results

debug test results directory

debug results.xml path

Adjust docker image and use `store_test_results` in circleCI

debug cmake version issue

check cmake version

Use new base image for new cmake version to output test results to ui

Rerun with new docker image

Use `store_test_results` to upload and store test results in circleCI

qianxichen233 committed:
Merge pull request #3 from qianxichen233/circleci-setup-update

Add store_test_results upload and store test results to circleCI

Shihuihuang1103 commented:
Hi Professor, all comments above are addressed and this should be ready for a second review. Thank you!
Here is the link to the CircleCI run: [link](https://app.circleci.com/pipelines/circleci/89RxQ8gXukBs6pXwF2XEFo/LBJg2NqgUMKxynB5gJyqDg/58/workflows/83ff7535-e97d-4ded-b561-b1d67ee53ee4)
